internal/service/subscription.go

```diff
@@ -3387,3 +3390,226 @@ func (s *subscriptionService) ActivateIncompleteSubscription(ctx context.Context
 
 	return nil
 }
+
+// SubscriptionConfig represents the configuration for subscription auto-cancellation
+type SubscriptionConfig struct {
+	GracePeriodDays         int  `json:"grace_period_days"`
+	AutoCancellationEnabled bool `json:"auto_cancellation_enabled"`
+}
+
+// GetSubscriptionConfig retrieves the subscription configuration from settings
+func (s *subscriptionService) GetSubscriptionConfig(ctx context.Context) (*SubscriptionConfig, error) {
+	setting, err := s.SettingsRepo.GetByKey(ctx, "subscription_config")
+	if err != nil {
+		// If setting not found, return default
+		if ierr.IsNotFound(err) {
+			s.Logger.Infow("subscription config not found, returning default",
+				"tenant_id", types.GetTenantID(ctx), "environment_id", types.GetEnvironmentID(ctx))
+			return &SubscriptionConfig{
+				GracePeriodDays:         1,
+				AutoCancellationEnabled: false,
+			}, nil
+		}
+		return nil, err
+	}
+
+	// Extract configuration from the setting value
+	config := &SubscriptionConfig{
+		GracePeriodDays:         3,
+		AutoCancellationEnabled: false,
+	}
+
+	// Extract grace_period_days
+	if gracePeriodDaysRaw, exists := setting.Value["grace_period_days"]; exists {
+		switch v := gracePeriodDaysRaw.(type) {
+		case float64:
+			config.GracePeriodDays = int(v)
+		case int:
+			config.GracePeriodDays = v
+		}
+	}
```

Comment:
This should be in types/settings.go as this has nothing to do with subscriptions